scaladoc/resources/dotty_res/scripts/ux.js

```diff
@@ -20,6 +20,22 @@ window.addEventListener("DOMContentLoaded", () => {
     $(this).parent().toggleClass("expanded")
   });
 
+  const observer = new IntersectionObserver(entries => {
+    entries.forEach(entry => {
+      const id = entry.target.getAttribute('id');
+      if (entry.intersectionRatio > 0) {
+        document.querySelector(`#toc li a[href="#${id}"]`).parentElement.classList.add('active');
+      } else {
+        document.querySelector(`#toc li a[href="#${id}"]`).parentElement.classList.remove('active');
+      }
+    });
+  });
+
+
+  document.querySelectorAll('#content .anchor[href]').forEach((section) => {
+    observer.observe(section);
+  });
+
```

Comment:
Could be added to Ux.scala, but it's ok

Reply:
It would be problematic because Scala.js don't provide API for IntersectionObserver. 

We could of course define a facade but I don't think it's worthwhile.